source/common/http/filter_manager.cc

```diff
@@ -791,12 +796,34 @@ void FilterManager::decodeMetadata(ActiveStreamDecoderFilter* filter, MetadataMa
       (*entry)->getSavedRequestMetadata()->emplace_back(std::move(metadata_map_ptr));
       return;
     }
-
+    state_.filter_call_state_ |= FilterCallState::DecodeMetadata;
     FilterMetadataStatus status = (*entry)->handle_->decodeMetadata(metadata_map);
-    ASSERT(!hasPreparedLocalReply(), "Sending Local Reply from Metadata is not yet supported.");
+    state_.filter_call_state_ &= ~FilterCallState::DecodeMetadata;
```

Comment:
This doesn't appear parallel to how the other filter state modification works?

Reply:
I swear I was looking at a version which didn't &=| out state.  This looks fine.